types/mytime.go

```diff
@@ -70,8 +70,9 @@ func (t MysqlTime) Microsecond() int {
 func (t MysqlTime) Weekday() gotime.Weekday {
 	// TODO: Consider time_zone variable.
 	t1, err := t.GoTime(gotime.Local)
+	// allow invalid dates
 	if err != nil {
-		return 0
+		return t1.Weekday()
```

Comment:
```suggestion
		t1, _ := t.GoTime(gotime.Local)
		return t1.Weekday()
```
I think this is better.

Reply:
ok